packaging/technical-addon/Makefile

```diff
@@ -31,89 +40,130 @@ MODINPUT_CONFIG_GENERATOR := $(TOOLS_DIR)/modinput_config_generator
 ta-build-tools:
 	@echo "Building tools..."
 	@mkdir -p $(TOOLS_DIR)
+	go build -o $(MODINPUT_CONFIG_GENERATOR) ./cmd/modinput_config_generator
 
 .PHONY: test-ta-build-tools
 test-ta-build-tools: ta-build-tools
-	cd $(SOURCE_DIR) && go build -o $(TOOLS_DIR)/modinput_config_generator ./cmd/modinput_config_generator
-	$(MODINPUT_CONFIG_GENERATOR) -source-dir=$(SOURCE_DIR)/cmd/modinput_config_generator/internal/testdata -schema-name=Sample_Addon --build-dir=$(BUILD_DIR) || exit 1; \
+	$(MODINPUT_CONFIG_GENERATOR) -source-dir=$(ADDONS_SOURCE_DIR)/cmd/modinput_config_generator/internal/testdata -schema-name=Sample_Addon --build-dir=$(BUILD_DIR) || exit 1; \
 	mkdir -p $(BUILD_DIR)/Sample_Addon/$(PLATFORM)_$(SPLUNK_ARCH)/bin; \
-	cd $(SOURCE_DIR) && go build $(GO_BUILD_FLAGS) -o $(BUILD_DIR)/Sample_Addon/$(PLATFORM)_$(SPLUNK_ARCH)/bin/Sample_Addon ./cmd/modinput_config_generator/internal/testdata/pkg/sample_addon/runner || exit 1; \
+	go build $(GO_BUILD_FLAGS) -o $(BUILD_DIR)/Sample_Addon/$(PLATFORM)_$(SPLUNK_ARCH)/bin/Sample_Addon $(ADDONS_SOURCE_DIR)/cmd/modinput_config_generator/internal/testdata/pkg/sample_addon/runner || exit 1; \
```

Comment:
Probably unrelated to this PR, feel free to drop: When I see a target that starts `test-`, I assume it's simply for testing. Is there a reason this target has to always build first? Also, why is this building manually rather than using a `build-` target of some kind?

Reply:
That's fair. It would be helpful to me to either stub out the "build a test environment" to its own make target and call it here, or add a comment above this target to clarify.